lib/realtime/channels.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Realtime.Channels do
+  @moduledoc """
+  Handles Channel related operations
+  """
+
+  alias Realtime.Api.Channel
+  alias Realtime.Repo
+
+  import Ecto.Query
+
+  @doc """
+  Creates a channel in the tenant database using a given DBConnection
+  """
+  @spec create_channel(map(), DBConnection.t()) :: :ok
+  def create_channel(attrs, db_conection) do
+    channel = Channel.changeset(%Channel{}, attrs)
+    {query, args} = Repo.insert_query_from_changeset(channel)
+
+    Postgrex.query!(db_conection, query, args)
+    :ok
+  end
+
+  @doc """
+  Fetches a channel by name from the tenant database using a given DBConnection
+  """
+  @spec get_channel_by_name(DBConnection.t(), String.t()) :: Channel.t() | nil
+  def get_channel_by_name(db_conection, name) do
+    query = from c in Channel, where: c.name == ^name
+    {query, args} = Repo.to_sql(:all, query)
```

Comment:
Why is it necessary to write a query in a DSL and then convert it into standard SQL?

Reply:
That is the same `select * from realtime.channels where name = $1?`

Reply:
I think it's cool 😎

Reply:
Using the DSL ends up being less error prone and avoids plain sql in the code which improves readability.

Reply:
Sure, but we will need to call a bunch of functions and perform several lookups from ETS, which seems too much to me for a simple SQL select

Reply:
this is the base to then have other more complex queries. I'm also not that concerned for performance in this query at the moment since this and other information for the Authz effort will use the Contextual Cache module if we need more reads